templates/decays.cpp.in

```diff
@@ -0,0 +1,301 @@
+ * which calculate the decay rates for particles in the model.
+ *
+ * This file was generated at @DateAndTime@ with FlexibleSUSY
+ * @FlexibleSUSYVersion@ (git commit: @FlexibleSUSYGitCommit@) and SARAH @SARAHVersion@ .
+ */
+
+#include <cmath>
+
+#include <boost/range/algorithm.hpp>
+
+#include "@ModelName@_decays.hpp"
+#include "@ModelName@_info.hpp"
+// @todo: move one_loop_decay_diagrams back to src
+#include "@ModelName@_one_loop_decay_diagrams.hpp"
+#include "concatenate.hpp"
+#include "dilog.hpp"
+#include "config.h"
+#include "thread_pool.hpp"
+#include "wrappers.hpp"
+#include "standard_model.hpp"
+
+namespace flexiblesusy {
+
+#define CLASSNAME @ModelName@_decays
+#define PHYSICAL(parameter) model.get_physical().parameter
+
+using namespace @ModelName@_cxx_diagrams;
+using namespace @ModelName@_cxx_diagrams::fields;
+namespace info = @ModelName@_info;
+using namespace std::complex_literals;
+
+const @ModelName@_decay_table& CLASSNAME::get_decay_table() const
+{
+   return decay_table;
+}
+
+void CLASSNAME::set_loop_corrections(const Decays_loop_corrections& loop_corrections_)
+{
+   loop_corrections = loop_corrections_;
+}
+
+const Decays_loop_corrections& CLASSNAME::get_loop_corrections() const
+{
+   return loop_corrections;
+}
+
+const Decays_problems& CLASSNAME::get_problems() const
+{
+   return problems;
+}
+
+void CLASSNAME::clear_problems()
+{
+   problems.clear();
+}
+
+void CLASSNAME::clear()
+{
+   clear_problems();
+   decay_table.clear();
+}
+
+void CLASSNAME::calculate_decays()
+{
+#ifdef ENABLE_THREADS
+@callAllDecaysFunctionsInThreads@
+#else
+@callAllDecaysFunctions@
+#endif
+}
+
+double get_alphas(context_base const&  context)
+{
+   return Sqr(context.model.get_@gs_name@())/(4.*Pi);
+}
+
+double get_alpha(context_base const&  context)
+{
+   return Sqr(unit_charge(context))/(4.*Pi);
+}
+
+/* 1-loop BSM amplitudes
+ *
+ * Notes:
+ *    - Since FS removes a factor of i from every vertex we reintroduce it
+ *      calls to one loop amplitudes
+ */
+
+@calcAmplitudeSpecializationDefs@
+
+// -------- specializations for decays needing higher order SM corrections --------
+
+@create_SM_particle_usings@
+
+namespace {
+
+unsigned int number_of_active_flavours(double m)
+{
+   /*
+   if (m < qedqcd.displayMbMb(){
+      return 4.0;
+   } else if (m < qedqcd.displayPoleMt()) {
+      return 5.0;
+   } else {
+      return 6.0;
+   }
+   */
+   return 5;
+}
+
+// eq. 2.7 of hep-ph/0503173
+double calc_A(double b)
+{
+   return (1 + b * b) *
+             (4. * dilog((1 - b) / (1 + b)) + 2. * dilog(-(1 - b) / (1 + b)) -
+              3. * std::log((1 + b) / (1 - b)) * std::log(2 / (1 + b)) -
+              2. * std::log((1 + b) / (1 - b)) * std::log(b)) -
+          3. * b * std::log(4. / (1 - b * b)) - 4. * b * std::log(b);
+}
+
+// eq. 2.6 of hep-ph/0503173
+double calc_DeltaH(double b)
+{
+   return calc_A(b)/b + 1./(16*b*b*b) * (
+      3 + 34*b*b - 13*Power4(b)) * log((1+b)/(1-b)) + 3./(8.*b*b)*(7*b*b-1);
```

Comment:
Please replace `log` by `std::log`.

Reply:
Done in 0908f6d4620ba035445c492b5c0d13a87f9738c5.